core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -70,22 +70,10 @@ object RequestChannel extends Logging {
                 memoryPool: MemoryPool,
                 @volatile private var buffer: ByteBuffer,
                 metrics: RequestChannel.Metrics) extends BaseRequest {
-    // These need to be volatile because the readers are in the network thread and the writers are in the request
-    // handler threads or the purgatory threads
-    @volatile var requestDequeueTimeNanos = -1L
-    @volatile var apiLocalCompleteTimeNanos = -1L
-    @volatile var responseCompleteTimeNanos = -1L
-    @volatile var responseDequeueTimeNanos = -1L
-    @volatile var apiRemoteCompleteTimeNanos = -1L
-    @volatile var messageConversionsTimeNanos = 0L
-    @volatile var temporaryMemoryBytes = 0L
-    @volatile var recordNetworkThreadTimeCallback: Option[Long => Unit] = None
-
     val session = Session(context.principal, context.clientAddress)
     private val bodyAndSize: RequestAndSize = context.parseRequest(buffer)
```

Comment:
We could take the chance and make this a `lazy val` so that we only parse the request when it's needed. In the common case, we would delay parsing until the request is being processed in `KafkaApis`.